openedx/features/redhouse_panel/api/v0/serializers.py

```diff
@@ -9,7 +21,98 @@ class SiteSerializer(serializers.Serializer):
     address = serializers.CharField()
 
 
-class UserSerializer(serializers.ModelSerializer):
+class UserProfileSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = UserProfile
+        fields = ('name', 'year_of_birth')
+
+
+class UserAccountSerializer(serializers.ModelSerializer):
+    profile = UserProfileSerializer()
+    can_access_panel = serializers.SerializerMethodField('has_panel_permission')
+    is_instructor = serializers.SerializerMethodField('can_create_courses')
+
     class Meta:
         model = User
-        fields = ('id', 'first_name', 'last_name', 'is_active')
+        read_only_fields = ('id',)
+        fields = ('username', 'email', 'is_active', 'profile', 'can_access_panel', 'is_instructor')
+
+    def has_panel_permission(self, obj):
+        if obj.groups.filter(name__in=[REDHOUSE_PANEL_GROUP_NAME]).exists():
+            return True
+        return False
```

Comment:
let's make a common method instead of this group checking everywhere.

Reply:
Done